### PR TITLE
Feature/paginate granted list

### PR DESCRIPTION
**Pull request by cedric25**
The list of authorized adresses for a specific protected data is now paginated.
It is now done with a Material UI DataGrid.
Addresses per page: 10

Before:
Capture d’écran 2023-10-31 à 17 12 04

After:
Capture d’écran 2023-10-31 à 17 13 06

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,10 +46,12 @@
     "eslint-config-airbnb-typescript": "^17.1.0",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-cypress": "^2.15.1",
+    "eslint-plugin-import": "^2.29.0",
     "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-sonarjs": "^0.21.0",
+    "prettier": "^2.8.4",
```
Comment (cedric25) [inline package.json:53]: Moved from dependencies to devDependencies 👍

File: package.json [hunk 2]
```diff
@@ -47,10 +46,12 @@
...
     "typescript": "^5.0.2",
-    "vite": "^4.3.2"
+    "vite": "^4.3.9"
```
Comment (cedric25) [inline package.json:55]: Bump `vite` to avoid vulnerable version.

File: src/app/appSlice.ts [hunk 3]
```diff
@@ -114,49 +116,73 @@ export const homeApi = api.injectEndpoints({
...
-                type: 'GRANTED_ACCESS' as const,
-                id: address,
-              })),
-              'GRANTED_ACCESS',
-            ]
-          : ['GRANTED_ACCESS'],
+      providesTags: (result) => {
+        if (!result?.grantedAccessList) {
+          return ['GRANTED_ACCESS'];
+        }
```
Comment (cedric25) [inline src/app/appSlice.ts:154]: Early break if no `grantedAccessList`.

File: src/features/consent/AuthorizedUsersList.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
+import { useState } from 'react';
```
Comment (cedric25) [inline src/features/consent/AuthorizedUsersList.tsx:1]: Component renamed from `ToggleList.tsx`

File: src/features/consent/Consent.tsx [hunk 5]
```diff
@@ -53,40 +65,46 @@ export default function Consent() {
...
+
       <Box id="summary">
-        <ul>
-          <li>
-            <h6>Owned by {protectedDataSelected?.owner}</h6>
-          </li>
-          <li>
-            <h6>Data Protected Address: {protectedDataSelected?.address}</h6>
-          </li>
+        <ul
```
Comment (cedric25) [inline src/features/consent/Consent.tsx:70]: Minor change to UI: Make items more readable = slightly bigger font size than before.

File: src/features/consent/GrantAccessModal.css [hunk 6]
```diff
@@ -3,7 +3,7 @@
   top: 50%;
   left: 50%;
   transform: translate(-50%, -50%);
-  width: 400px;
+  width: 500px;
```
Comment (cedric25) [inline src/features/consent/GrantAccessModal.css:6]: Minimum width for a wallet address.

File: src/features/consent/GrantAccessModal.css [hunk 7]
```diff
@@ -13,7 +13,3 @@
   align-items: center;
   gap: 16px;
 }
-
-#modalTitle {
-  text-align: left;
-}
```
Comment (cedric25) [inline src/features/consent/GrantAccessModal.css:19]: Clean dead code 🗑️

File: src/features/emailDapp/EmailDapp.tsx [hunk 8]
```diff
@@ -97,9 +97,9 @@ export default function EmailDapp() {
     <Box sx={{ m: 10, mx: 20 }}>
       <h2>Contact List</h2>
       <p>
-      Contacts who have protected their email address data and have allowed you to use it. 
-You can send them an email, without having access to their email address.
-
+        Contacts who have protected their email address data and have allowed
+        you to use it. You can send them an email, without having access to
+        their email address.
```
Comment (cedric25) [inline src/features/emailDapp/EmailDapp.tsx:102]: Prettier...

File: src/features/navigation/Navigation.tsx [hunk 9]
```diff
@@ -26,7 +26,7 @@ export default function Navigation() {
   const dispatch = useAppDispatch();
 
   const match = useMatch(`/:currentTab/*`);
-  const currentTab = match?.params.currentTab;
+  const currentTab = match?.params.currentTab ?? PROTECTED_DATA;
```
Comment (cedric25) [inline src/features/navigation/Navigation.tsx:29]: Avoid console error when `currentTab` is stil undefined.

File: package.json [hunk 10]
```diff
@@ -47,10 +46,12 @@
     "eslint-config-airbnb-typescript": "^17.1.0",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-cypress": "^2.15.1",
+    "eslint-plugin-import": "^2.29.0",
     "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-sonarjs": "^0.21.0",
+    "prettier": "^2.8.8",
```
Comment (cedric25) [inline package.json:53]: Moved from dependencies to devDependencies 👍

File: package.json [hunk 11]
```diff
@@ -47,10 +46,12 @@
...
     "typescript": "^5.0.2",
-    "vite": "^4.3.2"
+    "vite": "^4.5.0"
```
Comment (cedric25) [inline package.json:55]: Bump vite to avoid vulnerable version...

File: src/main.tsx [hunk 12]
```diff
@@ -7,8 +7,7 @@ import { WagmiConfig } from 'wagmi';
 import { Web3Modal } from '@web3modal/react';
 import { BrowserRouter } from 'react-router-dom';
 import { Provider } from 'react-redux';
-import { store } from './app/store';
-import { ethereumClient, projectId, wagmiClient } from './app/store';
+import { store, ethereumClient, projectId, wagmiClient } from './app/store';
```
Comment (cedric25) [inline src/main.tsx:10]: Only one import line `from './app/store'`
cf. ESLint rule "no-duplicate-imports"

File: src/features/consent/GrantAccessModal.tsx [hunk 13]
```diff
@@ -120,7 +129,11 @@ export default function GrantAcessModal(props: GrantAccessModalParams) {
             onChange={handleNbOfAccessChange}
             sx={{ mt: 3 }}
           />
-          <Button onClick={handleGrantAccess}>Validate</Button>
+          {/* TODO: Have a proper form and submit button */}
+          {/*<button type="submit">Validate</button>*/}
+          <Button disabled={result.isLoading} onClick={handleGrantAccess}>
+            {result.isLoading ? 'Loading...' : 'Validate'}
+          </Button>
```
Comment (cedric25) [inline src/features/consent/GrantAccessModal.tsx:136]: Disable button when loading state.

File: src/features/consent/GrantAccessModal.tsx [hunk 14]
```diff
@@ -7,18 +7,18 @@ import {
...
+import './GrantAccessModal.css';
 
 type GrantAccessModalParams = {
   protectedData: string;
   open: boolean;
   handleClose: () => void;
 };
 
-export default function GrantAcessModal(props: GrantAccessModalParams) {
+export default function GrantAccessModal(props: GrantAccessModalParams) {
```
Comment (cedric25) [inline src/features/consent/GrantAccessModal.tsx:21]: Fix typo in file name and here

File: src/features/consent/Consent.tsx [hunk 15]
```diff
@@ -53,40 +65,46 @@ export default function Consent() {
...
           <Box>
             <h2>1 to 1 messaging</h2>
-            <ToggleList authorizedUser={grantedAccessList} />
+            <AuthorizedUsersList
+              authorizedUsers={grantedAccessList}
+              count={count}
+              pageSize={AUTHORIZED_ADDRESSES_PER_PAGE}
+              page={page}
+              onPageChanged={(newPage: number) => setPage(newPage)}
+            />
```
Comment (cedric25) [inline src/features/consent/Consent.tsx:91]: The list of authorized users is now paginated, so it needs more props...

File: src/app/appSlice.ts [hunk 16]
```diff
@@ -114,49 +116,73 @@ export const homeApi = api.injectEndpoints({
...
     revokeOneAccess: builder.mutation<
       RevokedAccess | null,
       { protectedData: string; authorizedUser: string }
     >({
       queryFn: async (args) => {
         try {
-          const grantedAccessList =
-            await iExecDataProtector?.fetchGrantedAccess({
-              ...args,
+          const { protectedData, authorizedUser } = args;
```
Comment (cedric25) [inline src/app/appSlice.ts:171]: Personal preference: Avoid spreading args `...args` and be more explicit about what's given to `fetchGrantedAccess()`

File: src/features/consent/AuthorizedUsersList.tsx [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+          paginationMode="server"
+          paginationModel={paginationModel}
+          pageSizeOptions={[pageSize]}
+          rowCount={count}
+          onPaginationModelChange={({ page: newPage }) =>
+            onPageChanged(newPage)
+          }
+          sx={{ border: 'none' }}
+          disableRowSelectionOnClick={true}
+        />
```
Comment (cedric25) [inline src/features/consent/AuthorizedUsersList.tsx:108]: DataGrid pagination docs: https://mui.com/x/react-data-grid/pagination/

**Review by luisccastillo — APPROVED** [review]
Approved :heavy_check_mark:

**Pull request merged**